### ISSUE TITLE
File Provider errors out on middleware listing 

### ISSUE THREAD

**Issue by dalanmiller** [containous/traefik#4924]
Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [X] No

### What did you do?

Configured the middlewares like so:

```
[http.routers]
  [http.routers.synology]
      entryPoints = ["https"]
      # middlewares = ["compress", "redirect", "forward-auth"]
      rule = "Host(`synology.sub.domain.com`)"
      service = "synology"
```

### What did you expect to see?

A working *file.Provider

### What did you see instead?

In the logs and in the rawJSON output from Traefik I can see that the file.Provider is not loading properly. Removing this `middlewares` definition makes the issue go away. 

It seems like either `EntryPoints` should match `Middlewares` or vice versa?  
https://github.com/containous/traefik/blob/v2.0/pkg/config/dyn_config.go#L16-L17

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
v2.0.0-alpha4
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
[global]
  Debug = true
  SendAnonymousUsage = false

[api]

[providers]
  [providers.file]
    watch=true
    
  [providers.docker]
    endpoint = "unix:///var/run/docker.sock"
    domain = "sub.domain.com"
    watch = true
    exposedByDefault = false

[entryPoints]
  [entrypoints.http]
    address = ":80"

  [entrypoints.https]
    address = ":443"

[http.routers]

  [http.routers.synology]
      entryPoints = ["https", "http"]
      # middlewares = ["compress", "redirect", "forward-auth"]
      rule = "Host(`synology.sub.domain.com`)"
      service = "synology"
    
  [http.middlewares]
    [http.middlewares.compress.compress]

    [http.middlewares.redirect.redirectScheme]
      scheme = "https"

    [http.middlewares.forward-auth.forwardAuth]
      address = "https://traefik-forward-auth:4181"
      authResponseHeaders = ["X-Forwarded-User"]
      
    [http.services.synology]
      [http.services.synology.LoadBalancer]
          [[http.services.synology.LoadBalancer.servers]]
            url = "http://192.168.1.60:5000"
            scheme = "http"
            weight= 1

[acme]
  acmeLogging = true
  email = "me@email.com"
  storage = "/acme.json"
  onHostRule = true
  entryPoint = "https"
  [acme.dnsChallenge]
    provider = "a_well_known_provider"

  [[acme.domains]]
    main = "sub.domain.com"
  [[acme.domains]]
    main = "*.sub.domain.com"

```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

With the `middlewares` included in config
```
time="2019-05-31T22:16:48Z" level=error msg="Cannot start the provider *file.Provider: toml: cannot load TOML value of type string into a Go slice"
```

**Comment by dalanmiller** [comment]
Also what kind of test framework does `traefik` use? Typically I see `_test.go` files in the same directory. Was trying to see where `toml` file validation is done.

**Comment by ldez** [comment]
hello,
with the TOML file you provided I do not reproduce the problem.

**Comment by dalanmiller** [comment]
Forgot to mention I'm starting traefik via `docker-compose`:

```
traefik:
    container_name: traefik
    image: traefik:v2.0
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /nfs/docker/traefik/traefik.toml:/traefik.toml
      - /nfs/docker/traefik/acme.json:/acme.json
      - /nfs/docker/traefik/access.log:/access.log
      # - /nfs/docker/traefik/traefik.log:/traefik.log
    restart: always
    env_file: ./traefik.env
    labels:
      - "traefik.enable=false"
    dns:
      - 127.0.0.1
      - 1.1.1.1
      - 1.0.0.1
```

**Comment by dduportal** [comment]
Hi @dalanmiller , do you have the same error if your `traefik.toml` is stored on the local host machine (instead of the `/nfs` directory which looks like a NFS mountpoint)?

**Comment by dalanmiller** [comment]
@dduportal good 👁 ! Can confirm that trying to spin up Traefik locally with a non-mounted `traefik.toml` file arrives at the same result. I believe there to be something wrong with the parsing of the `middlewares` declaration: 

```
      middlewares = ["compress", "redirect", "forward-auth"]
```

Also tried:

```
      middlewares = "compress, redirect, forward-auth"
```

Also tried the following, thinking the hyphen was an issue:

```
      middlewares = ["compress", "redirect"]

```

.... I found the problem! 

I had added another `http.router` and had defined `entryPoints` with solely a string value instead of an array with a single item. All set here.
